internal/ctlog/ctlog.go

```diff
@@ -793,6 +793,11 @@ func (l *Log) sequencePool(ctx context.Context, p *pool) (err error) {
 		return fmtErrorf("couldn't sign checkpoint: %w", err)
 	}
 	l.c.Log.DebugContext(ctx, "uploading checkpoint", "size", len(checkpoint))
+
+	if testingOnlyFailCommit {
+		return errors.Join(errFatal, errors.New("failing commit checkpoint for test"))
```

Comment:
Why not `fmt.Errorf("failing commit checkpoint for test: %w", errFatal)` ?

Reply:
It's kind of the other way around, `errFatal` is not the cause of `failing commit checkpoint for test`, although both would technically work with the `errors.Is` check.

Reply:
Fair enough. At work we prefer `fmt.Errorf("%w: %w", foo, bar)` in similar cases but both `foo` and `bar` all well-defined errors (not an ad-hoc `errors.New`). Thanks.

Reply:
I... somehow failed to consider %w as a prefix. Maybe I got stuck on the old proposal to make a trailing %v wrap errors, who knows. Anyway, updated my two uses of error.Join, we can update this one too.